File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/ReadUSSensorReplyMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class ReadUSSensorReplyMessage extends Message {	
	public static int MESSAGE_ID				= 211;

	public static int DISTANCE_FIELD_LENGTH	= 4;
	
	private int distance;
	
	public int getDistance(){
		return distance;
	}
	
	public ReadUSSensorReplyMessage(int to, int from, int distance){
		super(MESSAGE_ID, 1, to, from);
		this.distance	= distance;
	}
	
	public ReadUSSensorReplyMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength() + DISTANCE_FIELD_LENGTH;
	}
	
	
	@Override
	protected void readData(String inputString){
		int startIndex		= 0;
		int endIndex		= DISTANCE_FIELD_LENGTH;
		distance			= Integer.parseInt(inputString.substring(startIndex, endIndex),10);
	}

	@Override
	protected String writeData(){
		return String.format("%04d", distance);
	}

	@Override
	protected boolean hasReply() {
		return false;
	}	
}

File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/ReadUSSensorReplyMessage.cpp
/*
 * ReadIRSensorsReplyMessage.cpp
 *
 *  Created on: Jun 24, 2013
 *      Author: mariano
 */

#include "ReadUSSensorReplyMessage.h"

namespace RobotProtocol {

ReadUSSensorReplyMessage ReadUSSensorReplyMessage::staticMessage = ReadUSSensorReplyMessage(0,0,0);

Message *ReadUSSensorReplyMessage::deserialize(char *serializedMessage){
	ReadUSSensorReplyMessage::staticMessage.initialize(serializedMessage);
	return &(ReadUSSensorReplyMessage::staticMessage);
}

ReadUSSensorReplyMessage::ReadUSSensorReplyMessage(int to, int from, int distance)
:Message(MESSAGE_ID, 1, to, from) {
	this->distance		= distance;
}

ReadUSSensorReplyMessage::ReadUSSensorReplyMessage(char *serializedMessage){
	initialize(serializedMessage);
}

ReadUSSensorReplyMessage::~ReadUSSensorReplyMessage() {
}

int ReadUSSensorReplyMessage::getMessageLength(){
	return Message::getMessageLength() + DISTANCE_FIELD_LENGTH;
}

int ReadUSSensorReplyMessage::getDistance(){
	return distance;
}

void ReadUSSensorReplyMessage::setDistance(int distance){
	this->distance		= distance;
}


void ReadUSSensorReplyMessage::readData(char **serializedMessageIndex){
	char *serializedMessage	= *serializedMessageIndex;
	char distanceField[DISTANCE_FIELD_LENGTH + 1];
	int index = 0;
	//int messageLength = (int)strtol(*serializedMessageIndex, &next, LENGTH_FIELD_LENGTH);
	distanceField[DISTANCE_FIELD_LENGTH] = '\0';
	memcpy(distanceField, &(serializedMessage[index]), DISTANCE_FIELD_LENGTH);
	distance = (int)strtol(distanceField, NULL, 10);
	index += DISTANCE_FIELD_LENGTH;
	(*serializedMessageIndex) += index;
}

void ReadUSSensorReplyMessage::writeData(char **serializedMessageIndex){
	sprintf(*serializedMessageIndex,"%04d", distance);
	(*serializedMessageIndex) = (*serializedMessageIndex) + DISTANCE_FIELD_LENGTH;
}

} /* namespace RobotProtocol */

